pkg/registry/apis/provisioning/register.go

```diff
@@ -355,18 +358,31 @@ func (b *APIBuilder) Validate(ctx context.Context, a admission.Attributes, o adm
 	}
 
 	list := ValidateRepository(repo)
+	cfg := repo.Config()
 
 	if a.GetOperation() == admission.Update {
-		cfg := repo.Config()
 		oldRepo, err := b.asRepository(ctx, a.GetOldObject())
 		if err != nil {
 			return fmt.Errorf("get old repository for update: %w", err)
 		}
+		oldCfg := oldRepo.Config()
 
-		if cfg.Spec.Type != oldRepo.Config().Spec.Type {
-			list = append(list, field.Invalid(field.NewPath("spec", "type"),
-				cfg.Spec.Type, "Changing repository type is not supported"))
+		if cfg.Spec.Type != oldCfg.Spec.Type {
+			list = append(list, field.Forbidden(field.NewPath("spec", "type"),
+				"Changing repository type is not supported"))
 		}
+
+		// Do not allow changing the sync target once anything has synced successfully
```

Comment:
If you enable sync, target should be mandatory in both, backend and frontend. I created it without `sync` from kubectl first and then enabled the sync via UI but target was not required or needed. 

![image](https://github.com/user-attachments/assets/3ed44002-0bab-4240-964a-d79746ec80e4)

Reply:
We should also block the change in that dropdown once is selected as we do for the backend and for the repository type. 

![image](https://github.com/user-attachments/assets/bd77b323-a210-4691-bc6d-81e6495482b3)

![image](https://github.com/user-attachments/assets/c000f707-be0e-44c2-a886-b962ab7d141a)


Btw, the name of that `managed folder` doesn't match the "mirror" naming we picked in the spec :(

Reply:
maybe: 
<img width="252" alt="image" src="https://github.com/user-attachments/assets/c2020323-a670-4f3d-90c7-76ae54eb80e3" />